accessibility_api/accessibility_lib/wrappers/IAccessible.py

```diff
@@ -0,0 +1,191 @@
+'''
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this file,
+You can obtain one at http://mozilla.org/MPL/2.0/.
+'''
+
+from ctypes import oledll, create_string_buffer
+from accessibility_api.accessibility_lib.wrappers.BaseAccessible import (
+    BaseAccessible
+)
+from accessibility_api.accessibility_lib.utils.WinUtil import WinUtil
+from ..scripts.constants import (
+    CHILDID_SELF, FULL_CHILD_TREE, SUCCESS, ERROR
+)
+from ..scripts.debug import DEBUG_ENABLED
+
+
+class IAccessible(BaseAccessible):
+    """
+    IAccessible windows interface
+    """
+
+    def __init__(self, params):
+        super(IAccessible, self).__init__(params)
+        # Find accessible object associated with ID
+        self._target = WinUtil.get_target_accessible(self.filtered_identifiers)
+
+    def serialize_result(self, depth=-1):
+        """
+        Return accessible object
+        """
+
+        if self._target is None:
+            return {
+                'error': ERROR,
+                'result': None
+            }
+        else:
+            return {
+                'error': SUCCESS,
+                'result': {'IAccessible': self.serialize(depth)},
+                'target': self._target,
+                'semantic_wrap': self.semantic_wrap
+            }
+
+    def serialize(self, child_depth=-1):
+        """
+        Serialize accessible into json
+        """
+
+        # Set child depth to full tree if none specified
+        if child_depth is None:
+            child_depth = -1
+        else:
+            child_depth = int(child_depth)
+
+        child_tree = {'Children': None}
+        parent_attrib = getattr(self._target, 'accParent')
+        role_attrib = getattr(self._target, 'accRole')
+        state_attrib = getattr(self._target, 'accState')
+        attributes = [
```

Comment:
Ok, I think what we need to organize here are 3 categories of fields:

getters (non-callable ones)
interface functions that need to be called
tree specific interface functions (children and parent, focus)

I think we should, and hopefully can, combine callable and non callable categories into 1. Every time we want to evaluate an attribute from the combined category, we can check if it is callable, and then call it.

Regarding the tree related callable ones, we should keep them in a separate list (since afaik, they are used in the serialize method).

Both lists should be declared as constants outside of the class and i think this way we can make it more readable. The non-tree related attributes should be common between serialize and semantic_wrap. The props we return everywhere should be the same.